debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnector.java

```diff
@@ -216,17 +216,18 @@ public Config validate(Map<String, String> connectorConfigs) {
         if (userValue.errorMessages().isEmpty()
                 && passwordValue.errorMessages().isEmpty()
                 && connectionStringValue.errorMessages().isEmpty()) {
-
             // Try to connect to the database ...
             ConnectionContext connContext = new ConnectionContext(config);
             try (MongoClient client = connContext.connect()) {
-                client.listDatabaseNames();
+                client.listDatabaseNames().first(); // only when we try to fetch results a connection gets established
             }
             catch (MongoException e) {
-                connectionStringValue.addErrorMessage("Unable to connect: " + e.getMessage());
+                if (connectorConfigs.containsKey(MongoDbConnectorConfig.CONNECTION_STRING.name())) {
+                    connectionStringValue.addErrorMessage("Unable to connect: " + e.getMessage());
+                    results.put(MongoDbConnectorConfig.CONNECTION_STRING.name(), connectionStringValue);
+                }
```

Comment:
If this is something that needs to be done then this is likely an issue at many other places (e.g MysqlConnector, PostgresConnector)